BLEServer/BLEServer/BLEServer.cpp

```diff
@@ -454,6 +455,28 @@ int main(Array<String^>^ args) {
 		msg->Insert("advType", JsonValue::CreateStringValue(eventArgs->AdvertisementType.ToString()));
 		msg->Insert("localName", JsonValue::CreateStringValue(eventArgs->Advertisement->LocalName));
 
+		// Serialise the individual raw AD structures as arrays of bytes in JSON
+		// The AD structures concatenated should contain the entire raw payload of the advertisement
+		// The entire payload is sent as JSON just in case we need other stuff in the future
+		auto adStructures = ref new JsonArray();
```

Comment:
What does `ref` do here?

Reply:
It's something from Microsoft C++/CLI. The `^` notation denotes a managed pointer and the `ref` keyword is used to create them. Funnily enough, `auto adStructures = new JsonArray()` actually doesn't compile

Reply:
Like C++ needed more keywords